crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -2546,16 +2616,24 @@ impl AuthorityPerEpochStore {
             batch,
             end_of_publish_transactions,
             commit_has_deferred_txns,
+            is_dkg_completed,
         )?;
 
-        Ok((verified_certificates, notifications, lock, final_round))
+        Ok((
+            verified_certificates,
+            notifications,
+            lock,
+            reserved_randomness_round,
+            final_round,
+        ))
     }
 
     fn process_end_of_publish_transactions_and_reconfig(
         &self,
         write_batch: &mut DBBatch,
         transactions: &[VerifiedSequencedConsensusTransaction],
         commit_has_deferred_txns: bool,
+        is_dkg_completed: Option<bool>,
```

Comment:
I think that is more confusing, because the name suggests that the bool is merely indicating whether the function should wait for DKG to complete, when in fact we are telling the function whether DKG has completed.

Reply:
i see - `Option<bool>` is still kind of confusing though. Can you just make it a bool and say that it is vacuously true if the dkg isn't supposed to be running?

Reply:
Yeah you are right, I also felt a little awkward writing it. In any case it's gone now that deferral is back